pkg/KubeArmorAuditPolicy/api/v1/kubearmorauditpolicy_types.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:validation:Minimum:=1
+// +kubebuilder:validation:Maximum:=10
+type SeverityType int
+
+// +kubebuilder:validation:Pattern=([0-9]+p[0-9]+s|[0-9]+p[0-9]+m)
+type RateLimitType string
+
+type ProcessType string
+
+type EventType struct {
+	Syscall string `json:"syscall,omitempty"`
+	Kprobe  string `json:"kprobe,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	Rate RateLimitType `json:"rate,omitempty"`
+
+	// socket related arguments
+	// +kubebuilder:validation:Optional
+	Protocol  int    `json:"protocol,omitempty"`
```

Comment:
if possible, it would be better to have validation, something like it should be one of "ip", "tcp", "udp", "icmp", and "raw".